### DISCUSSION TITLE
Suggest jedi_language_server  instead of pyright for Python

### DISCUSSION THREAD

**Discussion by ChenZhongPu** [LazyVim/LazyVim#1113] [Ideas]
Currently, it uses `pyright` as the LSP for Python, however `pyright` can only provide few package information for *hover* action (`Shift-k`) in neovim.

<img width="358" alt="image" src="https://github.com/LazyVim/LazyVim/assets/2310958/45274bb9-9079-49dd-b9dd-3b66edf23ca6">

I suggest replacing it with `jedi_language_server`, which is better in neovim.

<img width="1072" alt="image" src="https://github.com/LazyVim/LazyVim/assets/2310958/73f909de-d210-41ed-a027-15219a771668">

**Comment by fredrikaverpil** [comment]
The python extras for LazyVim won't fit everyone. I suggest you copy it and make your own configuration that you are happy with. And now that it got set up with pyright, the people using it now will be really confused if it changes.

Either copy the extras lua config into your own config, or set up something completely custom, like [I did](https://github.com/fredrikaverpil/dotfiles/tree/main/nvim-lazyvim/lua/plugins).

**Reply by ChenZhongPu** [reply]
Yes, `pyright` also has an extra advantage: it can be used for static analysis. So a better configuration is:

```lua
  {
    "neovim/nvim-lspconfig",
    opts = {
      servers = {
        jedi_language_server = {},
      },
      setup = {
        pyright = function()
          require("lazyvim.util").on_attach(function(client, _)
            if client.name == "pyright" then
              -- disable hover in favor of jedi-language-server
              client.server_capabilities.hoverProvider = false
            end
          end)
        end,
      },
    },
  },
```

**Reply by ChenZhongPu** [reply]
`Lazyvim.on_attach` is deprecated, and it can be repalced with `.lsp.on_attach`.
